### ISSUE TITLE
World Regeneration

### ISSUE THREAD

**Issue by MithrandirCraft** [IntellectualSites/FastAsyncWorldEdit#1983]
### What feature do you want to see added?

The posibility of actually regenerating entire worlds, while preserving claims/regions from commonly used protection plugins such as (in my case) WorldGuard & GriefDefender.

This is a feature many people have asked for before, and I am sure is quite vital to any survival server (unless they wish to reset their maps entirely every version).

FAWE/WorldEdit currently already has the capability of regenerating chunks, and I believe some of it's features already respect protected areas (not sure about regening though).

### Are there any alternatives?

AutoSaveWorld: Has this feature, but it's an old plugin that only works up to 1.12.
CyberWorldReset: Due to it's faulty design, which is dependant on how vanilla loads worlds, fails to be capable of regenerating the default world, which is a major hassle and complicates things.

### Anything else?

I understand this feature may be considered outside of FAWE's scope, but still believe it's an essential to any server, and want to give it a try and see what you have to say about it.

**Comment by SirYwell** [comment]
I'm not sure if I understand your use-case correctly. Do you want to regenerate everything *outside* of protected regions or do you just want to keep the protected regions?

**Comment by MithrandirCraft** [comment]
> I'm not sure if I understand your use-case correctly. Do you want to regenerate everything _outside_ of protected regions or do you just want to keep the protected regions?

Everything outside the protected regions should be regenerated. Preserving what is contained within the protected areas as it was.

**Comment by SirYwell** [comment]
Okay, that makes sense. There is one major issue: FAWE can only use the existing world generator, means if there were changes to the world generator in a Minecraft update, regenerating will result in what the *new* generator provides. That will most likely lead to ugly cuts on the borders of regions that aren't regenerated.

In general, it would be possible to implement something like that, but I think it shouldn't be in FAWE directly, but rather a plugin depending on FAWE.

**Comment by MithrandirCraft** [comment]
>  That will most likely lead to ugly cuts on the borders of regions that aren't regenerated.

True. I've done some regens in the past, and that is always the case. However, this is usually much more noticeable if the seed changes, like 1.17 to 1.18 (but it's something my server asumes, and in our case is eventually regenerated over, once and if players move their stuff somewhere clean).

> In general, it would be possible to implement something like that, but I think it shouldn't be in FAWE directly, but rather a plugin depending on FAWE.

That's what I thought. I still wanted input on it though, because it's a really necesary tool that I can't believe doesn't exist yet at this stage.

I'd do it myself, but I don't have the knowledge to, it's a delicate matter, and should be accessible to everyone.

No one has done this properly yet. IMO it has high potential for a premium resource.
